### PR TITLE
Add support for NCrunch

### PR DESCRIPTION
**Pull request by ltrzesniewski**
[NCrunch](https://www.ncrunch.net/) is a test runner on steroids which uses its own build directory and unfortunately requires additional configuration for each project that uses custom build tasks.

This PR makes the ANTLR code generator work out of the box by declaring additional items when the project is built under NCrunch. These items are automatically copied to NCrunch's build directory ([docs here](https://www.ncrunch.net/documentation/considerations-and-constraints_implicit-file-dependencies)).

I hope you won't mind adding this, as it will make life easier for those that use ANTLR along with NCrunch.

### CODE REVIEW

File: runtime/CSharp/Antlr4BuildTasks/Antlr4.CodeGenerator.targets [hunk 1]
```diff
@@ -196,4 +196,12 @@
   </Choose>
 
   <Import Condition="'$(Antlr4IsSdkProject)' == 'True'" Project="Antlr4.CodeGenerator.DefaultItems.targets" />
+  
+  <ItemGroup Condition="'$(NCrunch)' == '1'">
+    <None Include="$(MSBuildThisFileDirectory)..\tools\**\*" />
+    <None Include="@(Antlr4)" />
```
Comment (sharwell) [inline runtime/CSharp/Antlr4BuildTasks/Antlr4.CodeGenerator.targets]: 💭 According to the linked docs, it sounds like these items are already included by NCrunch.

File: runtime/CSharp/Antlr4BuildTasks/Antlr4.CodeGenerator.targets [hunk 2]
```diff
@@ -196,4 +196,12 @@
   </Choose>
 
   <Import Condition="'$(Antlr4IsSdkProject)' == 'True'" Project="Antlr4.CodeGenerator.DefaultItems.targets" />
+  
+  <ItemGroup Condition="'$(NCrunch)' == '1'">
+    <None Include="$(MSBuildThisFileDirectory)..\tools\**\*" />
```
Comment (sharwell) [inline runtime/CSharp/Antlr4BuildTasks/Antlr4.CodeGenerator.targets]: ❔ Can you explain why is this part needed? Does NCrunch not work with NuGet packages?

File: runtime/CSharp/Antlr4BuildTasks/Antlr4.CodeGenerator.targets (hunk 1 above)
Comment (ltrzesniewski) [reply runtime/CSharp/Antlr4BuildTasks/Antlr4.CodeGenerator.targets]: Looks like NCrunch actually ignores items it doesn't recognize. I'll double-check tomorrow just to make sure.

File: runtime/CSharp/Antlr4BuildTasks/Antlr4.CodeGenerator.targets (hunk 2 above)
Comment (ltrzesniewski) [reply runtime/CSharp/Antlr4BuildTasks/Antlr4.CodeGenerator.targets]: It will only copy the `props`/`targets` files included by NuGet (and I also think it transitively copies the imported files). It doesn't automatically copy the contents of the `tools` directory in ANTLR's case.

This is only relevant for old-style projects that use packages.config. The `PackageReference` format seems to work fine as NCrunch will reference the files from the NuGet cache directly without copying its contents, although this *may* still fail if you use a test grid (I haven't tested this combination).

**Comment by ltrzesniewski** [comment]
BTW I tested this on two projects (old style with packages.config and SDK-based with `PackageReference`).
I used NCrunch both locally and using a test grid node.

I tried to make a self-contained future-proof zero-maintenance change.

File: runtime/CSharp/Antlr4BuildTasks/Antlr4.CodeGenerator.targets (hunk 1 above)
Comment (ltrzesniewski) [reply runtime/CSharp/Antlr4BuildTasks/Antlr4.CodeGenerator.targets]: Actually I just tested it again: NCrunch won't copy the g4 files if they're only referenced by `Antlr4` items.

File: runtime/CSharp/Antlr4BuildTasks/Antlr4.CodeGenerator.targets (hunk 2 above)
Comment (ltrzesniewski) [reply runtime/CSharp/Antlr4BuildTasks/Antlr4.CodeGenerator.targets]: Tested again without this line: the new project format works, but the old one doesn't.

Surprisingly (at least to me), the new project format also works with a grid node - NCrunch uses its own NuGet cache in `C:\Temp\.nuget` to restore packages.

File: runtime/CSharp/Antlr4BuildTasks/Antlr4.CodeGenerator.targets (hunk 1 above)
Comment (sharwell) [reply runtime/CSharp/Antlr4BuildTasks/Antlr4.CodeGenerator.targets]: Thanks, can you add a short comment here mentioning this?

File: runtime/CSharp/Antlr4BuildTasks/Antlr4.CodeGenerator.targets (hunk 2 above)
Comment (sharwell) [reply runtime/CSharp/Antlr4BuildTasks/Antlr4.CodeGenerator.targets]: ❔ Should we condition this on `'$(Antlr4IsSdkProject)' != 'True'`?

Can you place a short comment here so future readers understand the context?

**Comment by ltrzesniewski** [comment]
I added some comments as requested.

I also added `Condition="'$(Antlr4IsSdkProject)' != 'True'"` even though I don't really think it's necessary. It doesn't hurt though.

**Comment by sharwell** [comment]
Thanks!
